### PR TITLE
feat: Enable alphabetize for imports

### PR DESCRIPTION
**Pull request by jirimoravcik**
Based off https://github.com/apify/apify-core/blob/develop/src/app/.eslintrc.js#L40-L60

The [rule](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md) is auto-fixable by eslint, so we won't have to do too much manual work (hopefully :D)

Slack discussion for reference https://apifier.slack.com/archives/C01VBUV81UZ/p1688991614682009

### CODE REVIEW

**Review by B4nan — APPROVED** [review]
tried to enable this rule in crawlee repo, there is apparently one bug in the import grouping check, but we can live with that. most of the problems were autofixed, it only struggled with some type imports and with imports surrounded with `ts-ignore` comments, but we had just a few of those

the PR is here: https://github.com/apify/crawlee/pull/1998/files

let's hope this wont hurt us. i've seen import order (of local files) resulting in cycles that were not possible to resolve (so you end up with a symbol resolving to `undefined` on runtime). iirc it was mostly visible when bundling projects via webpack, if you do that with some of your projects, keep that in mind if you see some classes resolved to `undefined` :]

**Review by fnesveda — APPROVED** [review]

**Comment by fnesveda** [comment]
I wish there was also a rule to enforce the `node:` prefix, but it looks like we're not getting it any time soon: https://github.com/import-js/eslint-plugin-import/issues/2717

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@apify/eslint-config",
-  "version": "0.3.4",
+  "version": "0.3.5",
```
Comment (B4nan) [inline package.json]: maybe this change deserves a minor bump? it will produce quite a lot of errors in most of the codebases

Comment (jirimoravcik) [reply package.json]: yeah, true. went for `0.4.0`.

**Review by mtrunkat — CHANGES_REQUESTED** [review]
Could we also put there the part about the `apify-packages` grouping? It's only used in private repositories, so this won't affect any open source repo. It makes it cleaner. 

```
    "import/order": ["error", {
        "groups": ["builtin", "external", ["parent", "sibling"], "index", "object"],
        "alphabetize": { "order": "asc", "caseInsensitive": true },
        "newlines-between": "always",
        "pathGroups": [
            // There is probably some bug in meteor that doesn't respect topological order of packages
            // and doesn't refresh intl strings properly. Importing intl before other packages fixes it
            // but we might run into similar issues in the future.
            {
                "pattern": "@apify-packages/intl**",
                "group": "external",
                "position": "after"
            },
            {
                "pattern": "@apify-packages/**",
                "group": "external",
                "position": "after"
            }
        ],
        "pathGroupsExcludedImportTypes": ["builtin"]
    }],
```

or as the comment says it's probably bug in the meteor we don't really need this elsewhere?

**Comment by mtrunkat** [comment]
Plus one more thing, I'd really also add this rule from core repo:

```
    "newline/import-module": ["warn", {
        "minItems": 2,
        "tab": 2,
    }],
```

as by having imports spread per multiple lines, we beware of merge conflicts in many cases. But we can leave this one for another discussion.

**Comment by jirimoravcik** [comment]
> Could we also put there the part about the `apify-packages` grouping? It's only used in private repositories, so this won't affect any open source repo. It makes it cleaner.
> 
> ```
>     "import/order": ["error", {
>         "groups": ["builtin", "external", ["parent", "sibling"], "index", "object"],
>         "alphabetize": { "order": "asc", "caseInsensitive": true },
>         "newlines-between": "always",
>         "pathGroups": [
>             // There is probably some bug in meteor that doesn't respect topological order of packages
>             // and doesn't refresh intl strings properly. Importing intl before other packages fixes it
>             // but we might run into similar issues in the future.
>             {
>                 "pattern": "@apify-packages/intl**",
>                 "group": "external",
>                 "position": "after"
>             },
>             {
>                 "pattern": "@apify-packages/**",
>                 "group": "external",
>                 "position": "after"
>             }
>         ],
>         "pathGroupsExcludedImportTypes": ["builtin"]
>     }],
> ```
> 
> or as the comment says it's probably bug in the meteor we don't really need this elsewhere?

I think we don't need it elsewhere. I wanted to keep it as simple as possible. If it causes issues, we can always add the grouping...

**Comment by jirimoravcik** [comment]
> Plus one more thing, I'd really also add this rule from core repo:
> 
> ```
>     "newline/import-module": ["warn", {
>         "minItems": 2,
>         "tab": 2,
>     }],
> ```
> 
> as by having imports spread per multiple lines, we beware of merge conflicts in many cases. But we can leave this one for another discussion.

I'd be for adding this rule too.
What do @fnesveda @B4nan @gippy think?

**Review by gippy — APPROVED** [review]

**Comment by B4nan** [comment]
I don't like that very much, 2 import statements seem to be too few for such a rule (at least for our usage in crawlee and similar), and I generally dislike warnings in linter, as they can be ignored and pollute the logs for no reason (aka it can make logs worse to navigate, and won't force people to actually follow the rule).

Maybe something like 5-6 would make more sense, if you want to enforce line breaks on (pretty much) all the import statements (except the single ones), it sounds better to put this into your project instead.

We already enforce the line length which will force us to wrap too long import statements anyway, to me that's good enough, personally, I don't like to line break the imports so aggressively.

sidenote: it should be `tab: 4`, not `tab: 2`, right?

edit: but I can understand the point with merge conflicts...

**Comment by fnesveda** [comment]
I don't mind adding it, but does it help with merge conflicts really? With 5 lines of git diff context by default, you'd have to have like 10 lines between the changed imports so that you don't get a merge conflict anyway, that wouldn't happen in most cases.

**Comment by jirimoravcik** [comment]
Ok guys, thanks for the discussion. I'm going to merge this PR as is. We can discuss the other rules later, since we don't perfectly agree on it.

**Comment by Strajk** [comment]
when u say "eslint" 3 times, Pablo will appear 🦄

**Pull request merged**